libcaf_core/caf/flow/op/on_error_resume_next.test.cpp

```diff
@@ -108,25 +107,33 @@ SCENARIO("on_error_resume_next discards items when there is no demand") {
   GIVEN("an observable that is retried on `runtime_error`") {
     WHEN("the demand is met") {
       THEN("the observer stops receiving new items") {
-        auto src1 = caf::flow::multicaster<int>(coordinator());
         auto snk = make_passive_observer<int>();
         auto on_error_resume_next_predicate = [](const caf::error& what) {
           return what == sec::runtime_error;
         };
-        auto uut = src1.as_observable()
-                     .on_error_resume_next(
-                       on_error_resume_next_predicate,
-                       make_observable().iota(1).take(2).as_observable())
-                     .subscribe(snk->as_observer());
+        using impl_t = caf::flow::op::on_error_resume_next_sub<
+          int, decltype(on_error_resume_next_predicate)>;
+        auto on_error_resume = caf::flow::op::on_error_resume_next{
+          coordinator(),
+          make_observable().never<int>(),
+          std::move(on_error_resume_next_predicate),
+          make_observable().never<int>(),
+        };
+        auto uut = intrusive_ptr<impl_t>{static_cast<impl_t*>(
+          on_error_resume.subscribe(snk->as_observer()).ptr())};
```

Comment:
Do we really need to use the operator manually? I think something like this should work?

```cpp
        auto sub = src1.as_observable()
                     .on_error_resume_next(
                       on_error_resume_next_predicate,
                       make_observable().iota(1).take(2).as_observable())
                     .subscribe(snk->as_observer());
        auto* uut = dynamic_cast<impl_t*>(sub.ptr());
        require(uut != nullptr);
```